src/source/src/manager.rs

```diff
@@ -35,6 +37,8 @@ use crate::{
 };
 
 pub type SourceRef = Arc<SourceImpl>;
+pub type SourceDescAsyncClosure =
+    Box<dyn Fn() -> Pin<Box<dyn Future<Output = Result<SourceDesc>>>> + Send + Sync>;
```

Comment:
```suggestion
    Box<dyn FnOnce() -> Pin<Box<dyn Future<Output = Result<SourceDesc>>>> + Send + Sync>;
```

Reply:
Actually, it is expected to execute more than one time.

Reply:
Then you can’t move the variables into the closure. 😭

Reply:
Also, this seems not be the correct for table source under currently design of `TableSouceV2`.

Reply:
Or let's make it an `enum`? One variant for storing a factory function and another one for storing an initialized singleton.